python/tests/integration/arcticdb/version_store/test_snapshot.py

```diff
@@ -491,4 +498,343 @@ def assert_0_delete_marker(lib, storage):
+    snap1 = "snap1"
+    snap2 = "snap2"
+
+    df_0 = create_df_index_rownum(10, 0, 10)
+    df_1 = create_df_index_rownum(10, 10, 20)
+
+    lib.write(symbol, df_0)
+    lib.snapshot(snap1)
+    lib.write(symbol, df_1)
+    lib.snapshot(snap2)
+    versions_initial = sorted([v["version"] for v in lib.list_versions()])
+
+    assert sorted(lib.list_snapshots()) == [snap1, snap2] 
+    assert_frame_equal(df_1, lib.read(symbol).data)     
+    assert versions_initial == [0, 1]
+    assert len([ver for ver in lib.list_versions() if ver["deleted"]]) == 0
+
+    lib.delete_version(symbol, 1)
+
+    assert sorted(lib.list_snapshots()) == [snap1, snap2] 
+    # Version is still returned, but marked for deletion
+    # Latest, or ver 1st is delete ver 0 is not
+    assert [ver["deleted"] for ver in lib.list_versions()] == [True, False]
+    assert_frame_equal(df_0, lib.read(symbol).data)     
+    # Althought the version is deleted it is not physically deleted
+    # It can be read from the snapshot still
+    assert df_1.equals(lib.read(symbol, as_of=snap2).data) 
+
+    lib.delete_snapshot(snap2)
+
+    assert sorted(lib.list_snapshots()) == [snap1] 
+    assert sorted([v["version"] for v in lib.list_versions()]) == [0]
+    assert len([ver for ver in lib.list_versions() if not ver["deleted"]]) == 1
+    assert_frame_equal(df_0, lib.read(symbol).data)     
+
+    # Cannot read from deleted snapshot label
+    with pytest.raises(Exception):
+        data = lib.read(symbol, as_of=snap2).data 
+
+    
+def test_delete_snapshot_basic_flow_with_delete_prev_version(basic_store):
+    lib = basic_store
+    symbol = "sym1"
+    snap1 = "snap1"
+    snap2 = "snap2"
+
+    df_0 = create_df_index_rownum(10, 0, 10)
+    df_1 = create_df_index_rownum(10, 10, 20)
+    df_combined = pd.concat([df_0, df_1])
+
+    lib.write(symbol, df_0)
+    lib.snapshot(snap1)
+    lib.append(symbol, df_1)
+    lib.snapshot(snap2)
+    versions_initial = sorted([v["version"] for v in lib.list_versions()])
+
+    assert sorted(lib.list_snapshots()) == [snap1, snap2] 
+    assert_frame_equal(df_combined, lib.read(symbol).data)     
+    assert versions_initial == [0, 1]
+
+    lib.delete_version(symbol, 0)
+
+    assert sorted(lib.list_snapshots()) == [snap1, snap2] 
+    # list_versions() return the newest versions first
+    assert [ver["deleted"] for ver in lib.list_versions()] == [False, True]    
+    assert_frame_equal(df_combined, lib.read(symbol).data)     
+    # we can still read the version as it is pointed in snapshot
+    assert_frame_equal(df_0, lib.read(symbol, as_of=snap1).data)     
+
+    lib.delete_snapshot(snap1)
+
+    assert_frame_equal(df_combined, lib.read(symbol).data)     
+    assert sorted(lib.list_snapshots()) == [snap2] 
+    assert [ver["deleted"] for ver in lib.list_versions()] == [False]
+
+@pytest.mark.xfail(reason = "ArcticDB#1863")    
+def test_delete_snapshot_complex_flow_with_delete_multible_smbls(basic_store_tiny_segment_dynamic):
```

Comment:
This unit test has a lot of stuff in it. I think we should keep it but also have more specific/minimal test targeted at #1863

Reply:
indeed - added more specific comment